prow/genjobs/gen-private-jobs.sh

```diff
@@ -91,15 +91,29 @@ go run ./genjobs \
   --env BAZEL_BUILD_RBE_JOBS=0,GCS_BUILD_BUCKET=istio-private-build,GCS_ARTIFACTS_BUCKET=istio-private-artifacts,DOCKER_REPOSITORY=istio-prow-build/envoy,ENVOY_REPOSITORY=https://github.com/istio-private/envoy,ENVOY_PREFIX=envoy \
   --repo-whitelist proxy
 
+# istio/release-builder test jobs(s) - pre/postsubmit(s)
+go run ./genjobs \
+  "${COMMON_OPTS[@]}" \
+  --branches=release-1.4,master \
+  --job-type presubmit,postsubmit \
+  --repo-whitelist release-builder \
+  --job-blacklist build-release_release-builder_release-1.4_postsubmit,build-release_release-builder_postsubmit,dry-run_release-builder_release-1.4,dry-run_release-builder_release-1.4_postsubmit,dry-run_release-builder,dry-run_release-builder_postsubmit,publish-release_release-builder_release-1.4_postsubmit,publish-release_release-builder_postsubmit
+
 # istio/release-builder build jobs(s) - postsubmit(s)
 go run ./genjobs \
   "${COMMON_OPTS[@]}" \
-  --labels preset-override-envoy=true \
+  --branches=release-1.4,master \
+  --labels preset-override-envoy=true,preset-override-deps=release-1.4 \
+  --env PRERELEASE_DOCKER_HUB=gcr.io/istio-prow-build,GCS_BUCKET=istio-private-prerelease/prerelease \
   --job-type postsubmit \
-  --repo-whitelist release-builder
+  --repo-whitelist release-builder \
+  --job-whitelist build-release_release-builder_release-1.4_postsubmit,build-release_release-builder_postsubmit
 
-# istio/release-builder test jobs(s) - presubmit(s)
+# istio/release-builder publish jobs(s) - postsubmit(s)
 go run ./genjobs \
   "${COMMON_OPTS[@]}" \
-  --job-type presubmit \
-  --repo-whitelist release-builder
+  --branches=release-1.4,master \
+  --env SOURCE_GCS_BUCKET=istio-private-prerelease/prerelease \
```

Comment:
I think we don't need this job confusingly. The build step does a build + publish to staging repo. The publish just pulls a release from staging, and re-publishes it to the final output (including github this time).

In private we only need to build and push to staging, then we will do the github publish in public -- otherwise we have tons of complexity around getting all of the public credentials in private

Reply:
Okay, that makes. I update and remove this job.